src/vcpkg/base/system.cpp

```diff
@@ -221,6 +227,32 @@ namespace vcpkg
 #endif
     }
 
+    void print_environment_variables()
+    {
+        msg::write_unlocalized_text_to_stdout(Color::none, "The following environment variables are currently set:\n");
+#if defined(_WIN32)
+        auto free = [](LPWCH p) { FreeEnvironmentStringsW(p); };
+        std::unique_ptr<WCHAR, decltype(free)> env_block(GetEnvironmentStringsW(), free);
+
+        for (LPWCH i = env_block.get(); *i != L'\0'; ++i)
+        {
+            msg::write_unlocalized_text_to_stdout(Color::none, Strings::concat(Strings::to_utf8(i), '\n'));
+
+            while (*i != L'\0')
+            {
+                ++i;
+            }
+        }
```

Comment:
```suggestion
        for (LPWCH i = env_block.get();;)
        {
            auto this_length = strlen(i);
            if (this_length == 0) { break; }
            msg::write_unlocalized_text_to_stdout(Color::none, Strings::concat(Strings::to_utf8(i, this_length), '\n'));
            i += this_length + 1;
        }
```

I would like to avoid so many repeated strlens :)

Reply:
Hm if an variable has an empty string as value, this would break the loop.

Reply:
@autoantwort The behavior isn't changed from the original:

```
            while (*i != L'\0')
            {
                ++i;
            }
```

is `i += strlen(i)`.

and `*i != L'\0'` will stop on an empty string.

As far as I know the entries are going to be of the form `KEY=VALUE\0` though so I don't think it actually occurs?

Reply:
True my mistake. It was too long ago and I made some wrong assumptions while writing my answer 😅